sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -1324,6 +1324,59 @@ def test_all_estimators_all_public():
     run_tests_without_pytest()
 
 
+def test_estimator_checks_generator_strict_none():
+    est = next(_construct_instances(NuSVC))
+    expected_to_fail = _get_expected_failed_checks(est)
+    # If we don't pass strict, it should not appear in the xfail mark either
+    # This way the behaviour configured in pytest.ini takes precedence.
+    checks = estimator_checks_generator(
+        est,
+        legacy=True,
+        expected_failed_checks=expected_to_fail,
+        mark="xfail",
+    )
+    # making sure we use a class that has expected failures
+    assert len(expected_to_fail) > 0
+    # xfail'ed checks are wrapped in a ParameterSet, so below we extract
+    # the things we need via a bit of a crutch: len()
+    marked_checks = [c for c in checks if len(c) == 3]
+
+    for parameter_set in marked_checks:
+        (_, check), marks, _ = parameter_set
+        mark = marks[0]
+        assert "strict" not in mark.kwargs
```

Comment:
Maybe a bit simpler to read like this?
```suggestion
    marked_checks = [c for c in checks if hasattr(c, "marks")]

    for parameter_set in marked_checks:
        first_mark = parameter_set.marks[0]
        assert "strict" not in first_mark.kwargs
```

You can also change the test below if you agree.

I was also wondering whether how much it is guaranteed that the first mark in the pytest.xfail one, but this is probably good enough ...